src/edu/csus/ecs/pc2/core/model/Problem.java

```diff
@@ -2105,6 +2105,9 @@ public boolean isUsingSandbox() {
      * @return an element of {@link Problem.SandboxType}.
      */
     public SandboxType getSandboxType() {
+        if (sandboxType == null) {
```

Comment:
Fix this.  For good design, there should be only one getSandboxType (place where the sandbox type if fetched/got) 
that code/method should include the Problem sandbox type field.
Code was added into Problem so getSandboxType would never be null, we should use/take advantage of that.

Reply:
Hmmm...  I'm sure what you're asking for. You said "Fix this", and then "For good design there should be only one getSandboxType()..."   But there IS only one such method -- this one.  What exactly are you suggesting needs to be fixed -- that is, what fix are you expecting to be done?

Reply:
As stated the method should use the Problem sandbox type field via Problem.getSandboxType() not a local class field.
The sandbox type in the Problem should be authoritative.

Lots of moving parts, I assumed that getSandboxType() was already changed, but I remembered it wrong.
In #361 the method that was changed was ContestInformation.getTeamScoreboardDisplayFormat().
The same bug condition (NPE) will happen with the new sandbox Problem fields if the get accessors
are not updated to handle if the field is null (due to Serialization of values).    The following Problem
methods should be updated to handle if the field == null:  getSandboxProgramName, getSandboxCmdLine,
and getSandboxType so no NPEs can occur.
This can be in scope for this issues as a robustness fix, or a new issue can be created.

Reply:
Doug agreed this comment is moot (he was looking at the wrong file).